src/main/java/fr/insee/keycloak/provider/FranceConnectIdentityProvider.java

```diff
@@ -103,7 +104,23 @@ protected boolean verify(JWSInput jws) {
     @Override
     public BrokeredIdentityContext getFederatedIdentity(String response) {
         try {
-            return super.getFederatedIdentity(response);
+            BrokeredIdentityContext federatedIdentity = super.getFederatedIdentity(response);
+            // Code to add verification for eidas level
+            // See: https://github.com/InseeFr/Keycloak-FranceConnect/issues/29
+            JsonWebToken idToken = (JsonWebToken) federatedIdentity.getContextData().get("VALIDATED_ID_TOKEN");
+            if (idToken == null) {
+                throw new IdentityBrokerException("The id_token cannot be retrieved");
+            }
+            FranceConnectIdentityProviderConfig.EidasLevel acrReturned = FranceConnectIdentityProviderConfig.EidasLevel.getOrDefault((String) idToken.getOtherClaims().get("acr"), FranceConnectIdentityProviderConfig.EidasLevel.EIDAS1);
+            FranceConnectIdentityProviderConfig.EidasLevel acrRequested = getFranceConnectConfig().getEidasLevel();
+            if (acrReturned == null || acrRequested == null) {
```

Comment:
- Agreed that acrRequested cannot be null
- acrReturned should not be null (I believe that FC will returned this claim to us all the time). However, note that "acr" is OPTIONAL in the id_token in the OIDC standard: https://openid.net/specs/openid-connect-core-1_0.html
It is to avoid ugly NPE if FC decides to change or has bug (who knows)...